pkg/controller/s3/bucket.go

```diff
@@ -183,3 +189,9 @@ func (e *external) Delete(ctx context.Context, mg resource.Managed) error {
 	_, err := e.s3client.DeleteBucketRequest(&awss3.DeleteBucketInput{Bucket: aws.String(meta.GetExternalName(cr))}).Send(ctx)
 	return resource.Ignore(s3.IsNotFound, err)
 }
+
+func lateInit(mg *v1beta1.Bucket) {
+	if mg.Spec.ForProvider.ACL == nil {
+		mg.Spec.ForProvider.ACL = aws.String("private")
```

Comment:
Can we use `GetBucketAcl` request after `HeadBucketRequest` to get the current state from AWS and do late-init against that info?

Reply:
We can also default at the CRD level, which I think may be preferable.

Reply:
The reason I did it this way, as opposed to changing the CRD was just prevented any breaking changes to the spec. But I guess if someone had the ALC missing previously, it wouldn't have worked regardless.

Reply:
Hm, actually, upon further inspection of the docs, this seems like the intended behavior.

> You can use either a canned ACL or specify access permissions explicitly. You cannot do both.

So maybe we don't need to change anything for this

Reply:
Just to make sure I understand correctly, if we submit a bucket with no ACL then it is going to fail right?

Reply:
You can either use a canned ACL or individual permissions, in the conversation linked in this PR, the manifest didn't use either, which is why it failed

Reply:
@krishchow gotcha, sounds like we should just add a note in the field descriptions then

Reply:
All done now :) @hasheddan